File: coupon/coupon-api/src/main/java/dev/be/coupon/api/coupon/presentation/dto/CouponCreateRequest.java
package dev.be.coupon.api.coupon.presentation.dto;

import dev.be.coupon.domain.coupon.CouponDiscountType;
import dev.be.coupon.domain.coupon.CouponType;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public record CouponCreateRequest(
        String couponName,
        CouponType couponType,
        CouponDiscountType couponDiscountType,
        BigDecimal couponDiscountValue,
        int totalQuantity,
        LocalDateTime expiredAt
) {
}

File: coupon/coupon-api/src/main/kotlin/dev/be/coupon/api/coupon/presentation/dto/CouponCreateRequest.kt
package dev.be.coupon.api.coupon.presentation.dto

import dev.be.coupon.api.coupon.application.dto.CouponCreateCommand
import java.time.LocalDateTime
import java.util.*

data class CouponCreateRequest(
    val name: String,
    val type: String,
    val totalQuantity: Int,
    val validFrom: LocalDateTime,
    val validUntil: LocalDateTime
)

fun CouponCreateRequest.toCommand(userId: UUID): CouponCreateCommand {
    return CouponCreateCommand(userId, name, type, totalQuantity, validFrom, validUntil)
}
